src/gdcmodels/export/cli.py

```diff
@@ -0,0 +1,164 @@
+    """
+    return gdcmodels.get_es_models(vestigial_included=True)
+
+
+class VestigialDelta(deepdiff.Delta):
+    """A custom Delta object which only writes the items added to the dictionary."""
+
+    def dumps(self):
+        # NOTE: The vestigial data is only data which must be added to the current
+        #       dictionary.
+        if "dictionary_item_added" in self.diff:
+            return self.serializer({"dictionary_item_added": self.diff["dictionary_item_added"]})
+
+        return self.serializer({})
+
+
+def _write_files(
+    index_name: str,
+    doc_type: str,
+    mapping: esmodels.ESMapping,
+    delta: deepdiff.Delta,
+    settings: Mapping[str, Any],
+    descriptions: Optional[Mapping[str, Any]],
+) -> None:
+    """Write the various data points to their appropriate files in the models.
+
+    Args:
+        index_name: The index name of the associated with the data.
+        doc_type: The doc-type name of the associated with the data.
+        mapping: The mapping of the index.
+        delta: The delta object representing any vestigial data.
+        settings: The settings with which the index needs to be configured.
+        descriptions: Any optional descriptions associated with the index.
+    """
+    esmodels_dir = resources.files(esmodels)
+    mapping_dir = index_name if index_name == doc_type else f"{index_name}/{doc_type}"
+    mapping_file = esmodels_dir / mapping_dir / "mapping.yaml"
+    vestigial_file = esmodels_dir / mapping_dir / "vestigial.yaml"
+    settings_file = esmodels_dir / index_name / "settings.yaml"
+    descriptions_file = esmodels_dir / index_name / "descriptions.yaml"
+
+    with resources.as_file(settings_file) as file, open(file, "w") as f:
+        extraction_utils.dump_yaml(settings, f)
+
+    if delta.diff:
+        with resources.as_file(mapping_file) as file, open(file, "w") as f:
+            extraction_utils.dump_yaml(mapping, f)
+
+        with resources.as_file(vestigial_file) as file, open(file, "w") as f:
+            delta.dump(f)
+
+    if descriptions:
+        with resources.as_file(descriptions_file) as file, open(file, "w") as f:
+            extraction_utils.dump_yaml(descriptions, f)
+
+
+def run_export(index_name: str, doc_type: str) -> None:
+    """Run the export of the index/doc-type.
+
+    Args:
+        index_name: The name of the index to export.
+        doc_type: The doc type associated with the index.
+    """
+    index = load_models()[index_name]
+    old_mapping, old_settings = index[doc_type]["_mapping"], index["_settings"]
+    _ = old_mapping.pop("_meta", None)
+
+    exporter = EXPORTERS[index_name][doc_type]
+    new_mapping, new_settings = exporter.export(old_mapping, old_settings)
+    descriptions = new_mapping.pop("_meta", {}).get("descriptions")
+
+    diff = deepdiff.DeepDiff(new_mapping, old_mapping)
+    delta = VestigialDelta(diff, serializer=extraction_utils.dump_yaml)
+
+    assert (new_mapping + delta) == old_mapping
```

Comment:
Should the `new_mapping + delta` _contain_ the `old_mapping` or exactly match it?

Reply:
It should be an exact match. Although we only _write_ the new fields that were added, this sanity check will apply the full delta including removing the newly created fields.